pontoon/api/tests/test_schema.py

```diff
@@ -0,0 +1,149 @@
+from django.test import TestCase
+from django.utils.encoding import force_text
+
+
+class TestCyclicQueries(TestCase):
+
+    def test_projects(self):
+        body = {
+            'query': '''{
+                projects {
+                    name
+                }
+            }'''
+        }
+
+        response = self.client.get('/graphql', body,
+                                   HTTP_ACCEPT="application/json")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(
+            force_text(response.content),
+            {"data": {"projects": [
+                {"name": "Pontoon Intro"}]}})
+
+    def test_projects_localizations(self):
+        body = {
+            'query': '''{
+                projects {
+                    localizations {
+                        locale {
+                            name
+                        }
+                    }
+                }
+            }'''
+        }
+
+        response = self.client.get('/graphql', body,
+                                   HTTP_ACCEPT="application/json")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(
+            force_text(response.content),
+            {'data': {'projects': [
+                {'localizations': [{'locale': {'name': 'English'}}]}]}})
+
+    def test_projects_localizations_cyclic(self):
+        body = {
+            'query': '''{
+                projects {
+                    localizations {
+                        locale {
+                            localizations {
+                                totalStrings
+                            }
+                        }
+                    }
+                }
+            }'''
+        }
+
+        response = self.client.get('/graphql', body,
+                                   HTTP_ACCEPT="application/json")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(
+            force_text(response.content),
+            {'data': {'projects': None},
+             'errors': [
+                 {'locations': [{'column': 17, 'line': 2}],
```

Comment:
This might be too specific. If you update the library and they change the format of this response, you'll have to update the test. I think here you can be fine with only verifying if ``Cyclic queries are forbidden`` is in the response content.